### PR TITLE
Class Overview

### LINKED ISSUE #49: Build Class Overview Page
Create a page that gives instructional members information about their class. The main feature of this page should be a table which shows a class's members

Later on this table will have functionality to edit a user's reputation, see details about a member, and an "add student" button.

### PR DESCRIPTION
**Pull request by DylanHalstead**
Closes #49

# Problem
We need Instructional Teams to have a way of seeing who is in their class, if they're restricted, and can restrict users is they choose too. The data for a section may change (like max bookable hours), so make it easy for the class to update this as well.

_Note that this has very little logic, and is mainly visual. The updating of user and section details will be another PR_

# Solution
A new page has been made at `src/routes/(signed_in)/courses/[courseID]/sections/[sectionID]/+page.svelte` which contains a section overview with a settings modal, and a table with all members of a section. Each row in this table how a details area where you can see, and later update, aspects of a user:

https://github.com/DylanHalstead/ClassConnect/assets/70990184/549d2be4-2b5c-4cf4-b3ad-7bc81eeeeb7f

**add users button will need to be added in another PR**

### CODE REVIEW

File: src/routes/+layout.svelte [hunk 1]
```diff
@@ -1,11 +1,9 @@
 <script lang="ts">
+	import "../app.css";
 	import { getFlash } from "sveltekit-flash-message";
```
Comment (jadenPete) [inline src/routes/+layout.svelte:3]: Could we sort these imports? 👉👈

File: src/routes/(signed_in)/courses/[courseID]/sections/[sectionID]/+page.server.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import type { PageServerLoad, Actions } from "./$types";
+import { getExtendedSectionMembers } from "$lib/db/sectionMembers";
+import { getExtendedSection } from "$lib/db/section";
+import { verifyAuthentication, verifyUserIsApartOfInstructionalTeam } from "$lib/auth";
+import { error } from "@sveltejs/kit";
+
+export const load: PageServerLoad = async ({ locals, cookies, params }) => {
+	const { sectionID } = params;
+	verifyAuthentication(locals, cookies);
+	const userID: string | undefined = cookies.get("userID");
```
Comment (jadenPete) [inline src/routes/(signed_in)/courses/[courseID]/sections/[sectionID]/+page.server.ts:10]: Nit: Is this type annotation necessary?

File: src/routes/(signed_in)/courses/[courseID]/sections/[sectionID]/+page.server.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import type { PageServerLoad, Actions } from "./$types";
+import { getExtendedSectionMembers } from "$lib/db/sectionMembers";
+import { getExtendedSection } from "$lib/db/section";
+import { verifyAuthentication, verifyUserIsApartOfInstructionalTeam } from "$lib/auth";
+import { error } from "@sveltejs/kit";
+
+export const load: PageServerLoad = async ({ locals, cookies, params }) => {
+	const { sectionID } = params;
+	verifyAuthentication(locals, cookies);
```
Comment (jadenPete) [inline src/routes/(signed_in)/courses/[courseID]/sections/[sectionID]/+page.server.ts:9]: Why not have `verifyAuthentication` return the user ID?

File: src/lib/db/sectionMembers.ts [hunk 4]
```diff
@@ -17,3 +18,39 @@ export async function getUsersSectionMembers(userId: string): Promise<SectionMem
...
+		const res: QueryResult<SectionMember> = await client.query(query);
+		const sectionMembers = res.rows;
+		if (sectionMembers.length === 0) {
+			return undefined;
+		}
+		return sectionMembers;
+	});
+}
+
+export async function getExtendedSectionMembers(
```
Comment (jadenPete) [inline src/lib/db/sectionMembers.ts:38]: Could you make an `extendSectionMembers` function that transformers `SectionMember`s into `ExtendedSectionMember`s? That way, we wouldn't have to define `Flat*` types, we wouldn't have unwieldy queries, and we could reuse all of the code we have for getting courses, users, etc..

File: src/lib/db/sectionMembers.ts [hunk 5]
```diff
@@ -17,3 +18,39 @@ export async function getUsersSectionMembers(userId: string): Promise<SectionMem
...
+export async function getSectionMembers(sectionId: string): Promise<SectionMember[] | undefined> {
+	return withConnection(async client => {
+		const query: QueryConfig = {
+			text: "SELECT sm.id, sm.section_id, sm.user_id, sm.member_type, sm.is_restricted FROM section_members sm WHERE sm.section_id = $1",
+			values: [sectionId]
+		};
+
+		const res: QueryResult<SectionMember> = await client.query(query);
+		const sectionMembers = res.rows;
+		if (sectionMembers.length === 0) {
```
Comment (jadenPete) [inline src/lib/db/sectionMembers.ts:31]: Nit: Shouldn't we return an empty array if there are no section members?

File: src/lib/components/modal/Modal.svelte [hunk 6]
```diff
@@ -0,0 +1,16 @@
+<script lang="ts">
+	let modalDialog: HTMLDialogElement;
+
+	export function open() {
```
Comment (jadenPete) [inline src/lib/components/modal/Modal.svelte:4]: Could we instead have `Modal` accept an `open` boolean prop that we listen to changes for? Then, we could call `showModal` where necessary.

**Review by jadenPete — APPROVED** [review]

File: src/lib/components/modal/Modal.svelte (hunk 6 above)
Comment (jadenPete) [reply src/lib/components/modal/Modal.svelte:4]: Watch and learn:
```
export let isOpen: boolean;

let modal: HTMLDialogElement;
let wasOpen = isOpen;

$: {
	if (!wasOpen && isOpen) {
		modal.showModal();
	} else if (wasOpen && !isOpen) {
		modal.close();
	}
	
	wasOpen = isOpen;
}
```

File: src/routes/(signed_in)/courses/[courseID]/sections/[sectionID]/+page.server.ts (hunk 2 above)
Comment (DylanHalstead) [reply src/routes/(signed_in)/courses/[courseID]/sections/[sectionID]/+page.server.ts:10]: By default it sets it to `any` but the only options are actually `string | undefined`

File: src/routes/(signed_in)/courses/[courseID]/sections/[sectionID]/+page.server.ts (hunk 3 above)
Comment (DylanHalstead) [reply src/routes/(signed_in)/courses/[courseID]/sections/[sectionID]/+page.server.ts]: :kiss:

**Pull request merged**